src/server/game/Server/Packets/SpellPackets.cpp

```diff
@@ -90,6 +90,76 @@ WorldPacket const* WorldPackets::Spells::SendUnlearnSpells::Write()
     return &_worldPacket;
 }
 
+template<class T, class U>
+bool WorldPackets::Spells::SandboxScalingData::GenerateDataForUnits(T* /*attacker*/, U* /*target*/)
+{
+    return false;
+}
+
+template<>
+bool WorldPackets::Spells::SandboxScalingData::GenerateDataForUnits<Creature, Player>(Creature* attacker, Player* target)
+{
+    CreatureTemplate const* creatureTemplate = attacker->GetCreatureTemplate();
+
+    Type                    = TYPE_CREATURE_TO_PLAYER_DAMAGE;
+    PlayerItemLevel         = target->GetAverageItemLevel();
+    TargetLevel             = target->getLevel();
+    Expansion               = creatureTemplate->RequiredExpansion;
+    Class                   = creatureTemplate->unit_class;
+    TargetMinScalingLevel   = (uint8)creatureTemplate->levelScaling->MinLevel;
+    TargetMaxScalingLevel   = (uint8)creatureTemplate->levelScaling->MaxLevel;
+    TargetScalingLevelDelta = (int8)creatureTemplate->levelScaling->DeltaLevel;
+    return true;
+}
+
+template<>
+bool WorldPackets::Spells::SandboxScalingData::GenerateDataForUnits<Player, Creature>(Player* attacker, Creature* target)
+{
+    CreatureTemplate const* creatureTemplate = target->GetCreatureTemplate();
+
+    Type                    = TYPE_PLAYER_TO_CREATURE_DAMAGE;
+    PlayerItemLevel         = attacker->GetAverageItemLevel();
+    TargetLevel             = target->getLevel();
+    Expansion               = creatureTemplate->RequiredExpansion;
+    Class                   = creatureTemplate->unit_class;
+    TargetMinScalingLevel   = (uint8)creatureTemplate->levelScaling->MinLevel;
+    TargetMaxScalingLevel   = (uint8)creatureTemplate->levelScaling->MaxLevel;
+    return true;
+}
+
+template<>
+bool WorldPackets::Spells::SandboxScalingData::GenerateDataForUnits<Creature, Creature>(Creature* attacker, Creature* target)
+{
+    CreatureTemplate const* creatureTemplate = target->HasScalableLevels() ? attacker->GetCreatureTemplate() : target->GetCreatureTemplate();
+
+    Type                    = TYPE_CREATURE_TO_CREATURE_DAMAGE;
+    TargetLevel             = target->getLevel();
+    Expansion               = creatureTemplate->RequiredExpansion;
+    Class                   = creatureTemplate->unit_class;
+    TargetMinScalingLevel   = (uint8)creatureTemplate->levelScaling->MinLevel;
+    TargetMaxScalingLevel   = (uint8)creatureTemplate->levelScaling->MaxLevel;
+    TargetScalingLevelDelta = (int8)creatureTemplate->levelScaling->DeltaLevel;
+    return true;
+}
+
+template<>
+bool WorldPackets::Spells::SandboxScalingData::GenerateDataForUnits<Unit, Unit>(Unit* attacker, Unit* target)
+{
+    // No scalable levels for any of them
+    if (   (!attacker->ToCreature() || !attacker->ToCreature()->HasScalableLevels())
+        && (!target->ToCreature()   || !target->ToCreature()->HasScalableLevels()))
```

Comment:
Actually i don't know any better way.. if you have something in mind, let me know